File: Java-Programing/Frontend/src/services/authService.ts
import api from "./api";

export const authService = {
  login: async (data: any) => {
    const res = await api.post("/auth/login", data);
    return res.data;
  },

  googleLogin: async (data: any) => {
    const res = await api.post("/auth/google", data);
    return res.data;
  },
  
  register: async (data: any) => {
    const res = await api.post("/auth/register", data);
    return res.data;
  },
  
  logout: async () => {
    await api.post("/auth/logout");
  },
  
  refreshToken: async () => {
    const res = await api.post("/auth/refresh");
    return res.data;
  },

  forgotPassword: async (email: string) => {
    const res = await api.post("/auth/forgot-password", { email });
    return res.data;
  },

  resetPassword: async (data: any) => {
    const res = await api.post("/auth/reset-password", data);
    return res.data;
  }
};

File: Java-Programing/Backend/src/main/java/com/careermate/service/AuthService.java
package com.careermate.service;

import com.careermate.dto.auth.AuthResponse;
import com.careermate.dto.auth.LoginRequest;
import com.careermate.dto.auth.RegisterRequest;

/**
 * Auth business operations.
 */
public interface AuthService {

    AuthResponse login(LoginRequest request);

    AuthResponse register(RegisterRequest request);

    /**
     * Logout is stateless for JWT. Kept for frontend contract.
     */
    void logout();

    /**
     * Refresh token is not implemented (no refresh token storage in this scaffold).
     * Kept for frontend contract.
     */
    AuthResponse refresh();

    void forgotPassword(String email);

    void resetPassword(Object payload);

    AuthResponse googleLogin(Object payload);
}
